app/src/main/java/ch/ubique/notifyme/app/MainViewModel.java

```diff
@@ -96,15 +103,11 @@ public CheckInState getCheckInState() {
 
 	public void setCheckedIn(boolean checkedIn) {
 		if (checkInState != null) checkInState.setCheckedIn(checkedIn);
-		storage.setCurrentVenue(checkInState);
+		setCheckInState(checkInState);
 	}
 
-	public boolean isCheckedIn() {
-		if (checkInState == null) {
-			return false;
-		} else {
-			return checkInState.isCheckedIn();
-		}
+	public LiveData<Boolean> isCheckedIn() {
+		return isCheckedIn;
```

Comment:
Why a getter for this LiveData but not for the others?

Reply:
Good questions. The other MutableLiveDatas should have a getter as well such that only a LiveData is exposed. I have changed this.